code/modules/robotics/bot/secbot.dm

```diff
@@ -26,18 +26,34 @@
 	var/target_lastloc //Loc of target when arrested.
 	var/last_found //There's a delay
 	var/frustration = 0
-	emagged = 0 //Emagged Secbots view everyone as a criminal
+	emagged = 0 			// Emagged Secbots view everyone as a criminal
 	health = 25
-	var/idcheck = 1 //If false, all station IDs are authorized for weapons.
-	var/check_records = 1 //Does it check security records?
-	var/arrest_type = 0 //If true, don't handcuff
-	var/report_arrests = 0 //If true, report arrests over PDA messages.
+	var/idcheck = 1 		// If false, all station IDs are authorized for weapons.
+	var/check_records = 1 	// Does it check security records?
+	var/arrest_type = 0 	// If true, don't handcuff
+	var/report_arrests = 0 	// If true, report arrests over PDA messages.
 
 	var/botcard_access = "Head of Security" //Job access for doors.
-	var/hat = null //Add an overlay from bots/aibots.dmi with this state.  hats.
+	var/hat = null 			// Add an overlay from bots/aibots.dmi with this state.  hats.
+	var/hatplus = null		// Add an overlay from clothing/item_hats.dmi with this state. more hats.
+	var/hatloot = null		// Give back that hatplus on death
+
 	var/our_baton_type = /obj/item/baton/secbot
 	var/loot_baton_type = /obj/item/scrap
-	var/stun_type = "stun"
+	var/botgun = null		// Which gun does our robot wield?
+	var/stun_type = "stun"	
+							// The following get defined by botgun!
+	var/spread = 0			// How inaccurate is our robot?
+	var/refire = 0 SECONDS	// How quickly should our bot fire each shot in a burst?
+	var/proj = null			// What's our bot shooting? Defined by botgun!
+	var/shotcount = 0		// Set by burst, modded by emag state
+	var/burst = 0			// How many shots does our robot shoot in a burst?
```

Comment:
should be handled in projectile

Reply:
Done-ish. Shotcount still exists, but only really modifies how many times shoot() is called.